legend-engine-xt-persistence-grammar/src/main/java/org/finos/legend/engine/language/pure/dsl/persistence/compiler/toPureGraph/IPersistenceCompilerExtension.java

```diff
@@ -67,4 +69,9 @@ default List<Function2<Trigger, CompileContext, Root_meta_pure_persistence_metam
     {
         return Collections.emptyList();
     }
+
+    default ValidationRuleSet<PersistenceContext> getExtraValidationRuleset()
```

Comment:
Wonder if we should support passing both the protocol object and the compiled pure object, and include the pure context.  This will allow validations to be delegated to pure functions.

Edit - wonder if better to return the same type as the commonRules:  

`function meta::pure::persistence::validation::commonRules(): ValidationRuleSet<PersistenceContext>[1]`

Reply:
Good point -- added support for validating based on pure or protocol objects.